nntrainer/graph/network_graph.cpp

```diff
@@ -422,11 +422,34 @@ int NetworkGraph::realizeGraph() {
 }
 
 void NetworkGraph::setBatchSize(unsigned int batch_size) {
+  if (batch_size == this->batch_size)
+    return;
+
   this->batch_size = batch_size;
+  auto allocated = tensor_manager->isAllocated();
+
+  if (allocated)
+    tensor_manager->deallocateTensors();
+
   for (auto iter = cbegin(); iter != cend(); iter++) {
     (*iter)->setBatch(batch_size);
+    if ((*iter)->isRunContextAvailable()) {
+      const RunLayerContext &context = (*iter)->getRunContext();
+      // resize tensors spec
+      for (unsigned int idx = 0; idx < context.getNumTensors(); idx++) {
+        auto const &ts = context.getTensor(idx);
+        tensor_manager->setBatchSize(ts.getName(), ts.getDim().batch());
```

Comment:
Q. what's the difference between this and line 449?

Reply:
L441 sets the batch size for the individual tensors by name - this covers the tensors requested by the layers.
L449 sets batch size for the tensors whose behaviors are known - weights, gradients, inputs, outputs, and derivatives.

Reply:
okay